google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -66,8 +68,19 @@
 public class Query {
 
   final FirestoreImpl firestore;
+  final UserDataConverter converter;
   final QueryOptions options;
 
+  public static class QueryPartition {
+    public final Object[] startAt;
+    @Nullable public final Object[] endBefore;
+
+    public QueryPartition(Object[] startAt, @Nullable Object[] endBefore) {
+      this.startAt = startAt;
+      this.endBefore = endBefore;
+    }
```

Comment:
I think we can improve this new class to also be able to generate the query for users in addition to holding the start & end values. This class is created in the context of an existing query, so we should be able to reference that to ease creation. Perhaps something like:

```suggestion
    }
    
    public Query createQuery() {
      return this.query.startAt(startAt).endBefore(endBefore);
    }
```

Reply:
I originally wanted to return only the query, but we don't have any getters for the cursor. This is a good middle ground.